src/matrix.rs

```diff
@@ -35,6 +21,54 @@ pub struct Matrix4<T> {
     pub m41: T, pub m42: T, pub m43: T, pub m44: T,
 }
 
+impl<T: Float> Matrix4<T> {
+    pub fn new(
```

Comment:
Note: We discussed removing the generics from this entirely in the meeting. But we can merge this independently.

cc @glennw